Doc/tools/extensions/implementation_detail.py

```diff
@@ -0,0 +1,47 @@
+"""Support for marking up implementation details."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from docutils import nodes
+from sphinx.locale import _ as sphinx_gettext
```

Comment:
Perhaps keep this as `_`? I think that's quite well know from a translation point of view and easy to look up.

Reply:
The `_ as sphinx_gettext` pattern was introduced in #98437, this reuses that for consistency (also in several of the other extensions).